### PR TITLE
Progress bar widget

### PR DESCRIPTION
**Pull request by Songtronix**
This will add a very primitive progressbar:
image

What additionally can be done:
- [X] make it customizable by accepting colors.
- [ ] allow the use of a border
- [ ] optional label in the middle showing the percentage
-  [ ] make progressbar render on the web

### CODE REVIEW

File: native/src/widget/progressbar.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+/// ```
+/// # use iced_native::Progressbar;
+/// 
+/// let value = 50.0;
+/// Progressbar::new(0.0..=100.0, value);
+/// ```
+///
+/// ![Default Progressbar](https://user-images.githubusercontent.com/18618951/71662391-a316c200-2d51-11ea-9cef-52758cab85e3.png)
+#[allow(missing_debug_implementations)]
+pub struct Progressbar {
```
Comment (hecrj) [inline native/src/widget/progressbar.rs]: I think we should name it `ProgressBar` and also change the example name to `progress_bar`.

File: native/src/widget/progressbar.rs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+/// Your [renderer] will need to implement this trait before being
+/// able to use a [`Progressbar`] in your user interface.
+///
+/// [`Progressbar`]: struct.Progressbar.html
+/// [renderer]: ../../renderer/index.html
+pub trait Renderer: crate::Renderer {
+    /// Returns the height of the [`Progressbar`].
+    ///
+    /// [`Progressbar`]: struct.Progressbar.html
+    fn height(&self) -> u32;
```
Comment (hecrj) [inline native/src/widget/progressbar.rs]: Maybe we could make the `height` configurable at the widget level and make this a `default_height` instead.

File: native/src/widget/progressbar.rs (hunk 1 above)
Comment (Songtronix) [reply native/src/widget/progressbar.rs]: Sure, will change.

File: native/src/widget/progressbar.rs (hunk 2 above)
Comment (Songtronix) [reply native/src/widget/progressbar.rs]: :ok_hand:

File: native/src/widget/progressbar.rs (hunk 1 above)
Comment (Songtronix) [reply native/src/widget/progressbar.rs]: Should I rename the module ``progress_bar`` too?

Comment (hecrj) [reply native/src/widget/progressbar.rs]: Yes!

**Comment by Songtronix** [comment]
Hmm I can't get the tour to compile to wasm. Either my setup is messed up for some dependency isn't wasm compatible anymore.
```Rust
error[E0599]: no method named `as_sock` found for type `&std::net::TcpStream` in the current scope
   --> /home/songtronix/.cargo/registry/src/github.com-1ecc6299db9ec823/net2-0.2.33/src/ext.rs:855:25
    |
855 |         do_connect(self.as_sock(), addr)
    |                         ^^^^^^^ method not found in `&std::net::TcpStream`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `as_sock`, perhaps you need to implement it:
            candidate #1: `ext::AsSock`
```

**Comment by hecrj** [comment]
This is probably a dev dependency, maybe `surf`? Rust will try to compile all the dev dependencies even if they are not used by the example you are trying to build.

Although it would be great to use a cross-platform HTTP client, maybe we could make the dev dependency conditional based on the target for the time being.

**Comment by Songtronix** [comment]
Okay by disabling almost all dev dependencies helped. I'll try to get the progressbar working on the web.

**Comment by Songtronix** [comment]
Well implementing the progressbar for the web is harder than I thought, instead I've fixed the text_input for the web to support being a password field.

**Comment by Songtronix** [comment]
Good idea to move the default height over to the render. I'm fine with merging it :ok_hand:

**Pull request merged**